src/services/pallets/PalletsErrorsService.ts

```diff
@@ -0,0 +1,298 @@
+	palletId: string;
+}
+
+interface IFetchErrorItemArgs extends IFetchPalletArgs {
+	errorItemId: string;
+	metadata: boolean;
+}
+
+export class PalletsErrorService extends AbstractService {
+	async fetchErrorItem(
+		historicApi: ApiDecoration<'promise'>,
+		{ hash, palletId, errorItemId, metadata }: IFetchErrorItemArgs
+	): Promise<IPalletErrorsItem> {
+		const palletMetadata = historicApi.registry.metadata;
+
+		const [palletMeta, palletMetaIdx] = this.findPalletMeta(
+			palletMetadata,
+			historicApi,
+			palletId
+		);
+
+		const palletName = stringCamelCase(palletMeta.name);
+		const errors = historicApi.errors[palletName];
+
+		// Even if `errorItemMeta` is not used, we call this function to ensure it exists. The side effects
+		// of the error item not existing are that `findErrorItemMeta` will throw.
+		const errorItemMetadata = this.findErrorItemMeta(
+			palletMeta,
+			errorItemId,
+			errors
+		);
+
+		let normalizedErrorItemMeta;
+		if (metadata) {
+			normalizedErrorItemMeta = this.normalizeErrorItemMeta(
+				(errorItemMetadata[1] as unknown as IsError).meta
+			);
+		}
+
+		const [, { number }] = await Promise.all([
+			historicApi.query[palletName][errorItemId],
+			this.api.rpc.chain.getHeader(hash),
+		]);
+
+		return {
+			at: {
+				hash: hash,
+				height: number.unwrap().toString(10),
+			},
+			pallet: stringCamelCase(palletMeta.name),
+			palletIndex: palletMetaIdx,
+			errorItem: errorItemId,
+			metadata: normalizedErrorItemMeta,
+		};
+	}
+
+	async fetchErrors(
+		historicApi: ApiDecoration<'promise'>,
+		{ hash, palletId, onlyIds }: IFetchPalletArgs & { onlyIds: boolean }
+	): Promise<IPalletErrors> {
+		const metadata = historicApi.registry.metadata;
+		const [palletMeta, palletMetaIdx] = this.findPalletMeta(
+			metadata,
+			historicApi,
+			palletId
+		);
+
+		const { number } = await this.api.rpc.chain.getHeader(hash);
+		const errors = historicApi.errors[palletMeta.name.toString().toLowerCase()];
+
+		let items: [] | ErrorMetadataLatest[] | Text[];
+		if (palletMeta.errors.isNone) {
+			items = [];
+		} else if (onlyIds) {
+			items = Object.entries(errors).map(
+				(errorItem) => errorItem[0] as unknown as Text
+			);
+		} else {
+			items = [] as ErrorMetadataLatest[];
+			for (const [, value] of Object.entries(errors)) {
+				const item = value.meta;
+				items.push(item);
+			}
+		}
+
+		return {
+			at: {
+				hash: hash,
+				height: number.unwrap().toString(10),
+			},
+			pallet: stringCamelCase(palletMeta.name),
+			palletIndex: palletMetaIdx,
+			items,
+		};
+	}
+
+	/**
+	 * Normalize error item metadata by running it through `sanitizeNumbers` and
+	 * converting the docs section from an array of strings to a single string
+	 * joined with new line characters.
+	 *
+	 * @param errorItemMeta polkadot-js ErrorEntryMetadataV12
+	 */
+	private normalizeErrorItemMeta(
+		errorItemMeta: ErrorMetadataLatest
+	): ISanitizedErrorItemMetadata {
+		const normalizedErrorItemMeta = sanitizeNumbers(
```

Comment:
It's important to know for performance costs, `sanitizeNumbers` can be costly. Questions:

1. How large is the data we are passing in and does it really make a difference?

2. Is there a way to normalize it without calling `sanitizeNumbers`?

Reply:
After looking at the data a bit more I don't think it makes a difference. The `docs` field in the meta for errors seem to just be a description of the error so I think we can remove the call to sanitizeNumbers. Ill update this

Reply:
Awesome sounds good :)